bin/handler-slack-multichannel.rb

```diff
@@ -256,31 +263,44 @@ def post_data(notice, channel)
     http.use_ssl = true
 
     begin
-      req = Net::HTTP::Post.new("#{uri.path}?#{uri.query}")
-      if payload_template.nil?
-        text = slack_surround ? slack_surround + notice + slack_surround : notice
-        req.body = payload(text, channel).to_json
+      # Implement a retry strategy, with 5 tries max and a timeout of 10 seconds each
+      tries ||= 5
+      Timeout.timeout(10) do
+        begin
+          req = Net::HTTP::Post.new("#{uri.path}?#{uri.query}")
+          if payload_template.nil?
+            text = slack_surround ? slack_surround + notice + slack_surround : notice
+            req.body = payload(text, channel).to_json
+          else
+            req.body = notice
+          end
+          puts "request: #{req}"
+
+          response = http.request(req)
+
+          puts "response: #{response}"
+
+        rescue Net::HTTPServerException => error
+          if (tries -= 1) > 0
+            sleep 5
+            puts "retrying incident #{incident_key}... #{tries} left"
+            retry
+          else
+            # raise error for sensu-server to catch and log
+            puts 'slack api failed (retries) ' + incident_key + ' : ' + error.response.code + ' ' + error.response.message + ': sending to channel "' + channel + '" the message: ' + notice
+            exit 1
+          end
+        end
+      end
+    rescue Timeout::Error
+      if (tries -= 1) > 0
+        puts "retrying... #{tries} left"
+        retry
       else
-        req.body = notice
+        # raise error for sensu-server to catch and log
+        puts 'slack api failed (timeout) ' + incident_key + ' : sending to channel "' + channel + '" the message: ' + notice
```

Comment:
use string interpolation:
```suggestion
        puts "slack api failed (timeout) #{incident_key} : sending to channel #{channel} the message: #{notice}"
```

Reply:
Same as before